lua/terrortown/migrations/2024_02_16_23_10_favorites_migration.lua

```diff
@@ -0,0 +1,50 @@
+--- @ignore
+-- Migration file for favorites
+if SERVER then
+    return function() end
+end
+
+local oldDatabaseName = "ttt_bem_fav"
+local newDatabaseName = "ttt2_shop_favorites"
+local savingKeys = {}
+
+---
+-- Checks old favorites table and merges them into the new table
+-- @realm client
+return function()
+    local sqlCommand = "CREATE TABLE " .. newDatabaseName .. " (name TEXT PRIMARY KEY)"
+
+    if not sql.TableExists(newDatabaseName) and sql.Query(sqlCommand) == false then
+        error("[TTT2] Couldn't create sql table in migrations. Query failed.")
+    end
```

Comment:
i would split this up, as both failures a somewhat different.

if the table already exists (not created by us) we can tell the user this and allow them to delete the offending database table. Or even harsher we could claim this table (or rather all tables prefixed with `ttt2_` and drop or rename the table and go forward.
This would allow for a nicer errorhandling that doesn't result in a non-working gamemode.

The sql.Query could fail for various Reasons, if we take away the `table already exists` error there is no reasonable action we could take to make this work. Maybe we can get the sql error message and display it, but that's as far as it can go.